charts/istio-alerts/templates/service-prometheusrule.yaml

```diff
@@ -85,5 +85,37 @@ spec:
         {{ toYaml . | nindent 8}}
         {{- end }}
     {{- end }}
-  {{- end }}
+    {{- end }}
+
+  - name: {{ .Release.Name }}.{{ .Release.Namespace }}.{{ .Chart.Name }}.alertRulesSelectorsValidity
+    rules:
+    {{- with .Values.serviceRules.destinationServiceSelectorValidity }}
+    - alert: DestinationServiceSelectorValidity
+      annotations:
+        summary: >-
+          DestinationServiceSelector for istio-alerts is invalid
+        runbook_url: {{ $values.defaults.runbookUrl}}#Alert-Rules-Selectors-Validity
+        description: >-
+          The DaemonsetSelector used for daemonset level alerts did not return any data.
+          Please check the DaemonsetSelector applied in your prometheus-alerts chart
+          is correct to ensure you are properly selecting your daemonsets so that you
+          will be alerted for daemonset issues. The current selector is
+          `{destination_service_namespace="{{ $release.Namespace }}", destination_service_name="{{ $.Values.serviceRules.destinationServiceName }}"}`.
```

Comment:
So - I think we need to create functions now that define the selectors. We should create a function at the top of the file like this:

```
{{- define "istioAlerts.selectors" -}}
printf "destination_service_namespace=\"%s\", "destination_service_name=\"%s\"" $.Release.Namespace $.Values.serviceRules.destinationServiceName }}
{{- end }}
```

Then in each of the alerts, we can use that function... ie:

```promql
        histogram_quantile(
          {{ .percentile }},
          sum(irate(
            istio_request_duration_milliseconds_bucket{
              {{ include "istioAlerts.selectors" $ }}
            }[5m]
          )) by (
            destination_service_name,
            reporter,
            source_canonical_service,
            le
          )
        )
        / 1000
        > {{ .threshold }}
```

(we should do this in all of the alerts)

Reply:
I will do this in a separate small PR. skipping for now